extension/src/TestSingleton.cpp

```diff
@@ -6,31 +6,26 @@
 using namespace godot;
 using namespace OpenVic2;
 
-TestSingleton *TestSingleton::singleton = nullptr;
+TestSingleton* TestSingleton::singleton = nullptr;
```

Comment:
Foremost disagrees with https://github.com/OpenVic2Project/OpenVic2/blob/537938683b748dbf5fcd78276aa823f168f715f1/extension/src/TestSingleton.hpp#L11

Also the type is part of the variable declaration in C++, the pointer is attached to variable identifier so should be associated with the identifier over the type.

Reply:
>Format changes should be separated from this PR, format changes/fixes should be delegated to either a separate commit or PR, as it also reduces possibility of conflict by modifying unrelated files unnecessarily so.

>Also the type is part of the variable declaration in C++, the pointer is attached to variable identifier so should be associated with the identifier over the type.

Until the style guide is done, all formatting is considered temporary

>Should also separate declarations into headers and definitions into source files whenever possible.

In general, I would agree with this sentiment. However the purpose of the PR is to give a more advanced example of Godot UI interacting with C++ code, so it is in a single file for ease as an instructional tool

Reply:
> Until the style guide is done, all formatting is considered temporary

Format changes (to the main repo) should still be separate form any PR outside of a format fix PR, this should be enforced as a general guideline default for commits on all version control systems regardless of a styleguide (or a contribution guide either) as this is a very good way to introduce conflict to everyone else's forks without warning. If we have one single commit/PR for format corrections every so often (should only need it once but say this happens again) we can publicly point to it and when its merged have folks update and rebase to it, but if we do it in any PR whenever we can we will have many random format changes constantly.

> In general, I would agree with this sentiment. However the purpose of the PR is to give a more advanced example of Godot UI interacting with C++ code, so it is in a single file for ease as an instructional tool

Well foremost we should adhere to the principals of best formatting rules that make sense where possible while we lack a styleguide, so even beside my previous point regarding formatting commits, a best rule for this case would be to keep the pointer with the identifier, this would teach best practices because even the little details of formatting will become default for how people will write the code even before the styleguide is written. This should be properly addressed in the completed styleguide but default manner of writing should follow how the C++ parser will read it.

Reply:
Changes of commit e7ee0dd have been reverted for now